site/src/components/Modal/CreateSchemaModal.tsx

```diff
@@ -0,0 +1,140 @@
+  const router = useRouter();
+  const { user } = useUser();
+
+  // @todo might be a good idea to split this into a hook
+  // @see https://github.com/blockprotocol/blockprotocol/pull/223#discussion_r808072665
+  const handleSchemaTitleChange = (value: string) => {
+    let formattedText = value.trim();
+    // replace all empty spaces
+    formattedText = formattedText.replace(/\s/g, "");
+
+    // capitalize text
+    if (formattedText.length > 0) {
+      formattedText = formattedText[0].toUpperCase() + formattedText.slice(1);
+    }
+
+    setNewSchemaTitle(formattedText);
+  };
+
+  const handleCreateSchema = useCallback(
+    async (evt: FormEvent) => {
+      evt.preventDefault();
+      if (!user || user === "loading") {
+        return;
+      }
+
+      unstable_batchedUpdates(() => {
+        setTouchedInput(true);
+        setLoading(true);
+        setApiErrorMessage(undefined);
+      });
+
+      const { data, error } = await apiClient.createEntityType({
+        schema: {
+          title: newSchemaTitle,
+        },
+      });
+      setLoading(false);
+      if (error) {
+        setApiErrorMessage(error.message);
+      } else if (data) {
+        const schemaTitle = data.entityType.schema.title;
+        void router.push(`/@${user.shortname}/types/${schemaTitle}`);
+      }
+    },
+    [user, newSchemaTitle, router],
+  );
+
+  // @todo introduce a library for handling forms
+  const helperText = touchedInput
+    ? apiErrorMessage ||
+      (newSchemaTitle === "" ? "Please enter a valid value" : undefined)
+    : undefined;
+
+  const isSchemaTitleInvalid = !!apiErrorMessage || newSchemaTitle === "";
+
+  const displayError = touchedInput && isSchemaTitleInvalid;
+
+  return (
+    <Modal
+      open={open}
+      onClose={onClose}
+      contentStyle={{
+        top: "40%",
+      }}
+    >
+      <Box sx={{}}>
+        <Typography
+          variant="bpHeading4"
+          sx={{
+            mb: 2,
+            display: "block",
+          }}
+        >
+          Create New <strong>Schema</strong>
+        </Typography>
+        <Typography
+          sx={{
+            mb: 4,
+            fontSize: 16,
+            lineHeight: 1.5,
+            width: { xs: "90%", md: "85%" },
+          }}
+        >
+          {`Schemas are used to define the structure of entities - in other words,
+          define a ‘type’ of entity`}
+        </Typography>
+        <Box component="form" onSubmit={handleCreateSchema}>
+          <TextField
+            sx={{ mb: 3 }}
+            label="Schema Title"
+            fullWidth
+            helperText={helperText}
+            value={newSchemaTitle}
+            onChange={(evt) => {
+              if (apiErrorMessage) {
+                setApiErrorMessage(undefined);
+              }
+              handleSchemaTitleChange(evt.target.value);
+            }}
+            required
+            error={displayError}
+          />
+
+          <Button
+            disabled={displayError}
```

Comment:
This shouldn't just be disabled when an error is being displayed, it should be disabled when there is any error in the input. When the modal is loaded for the first time for instance, the button should be disabled because the input is empty. Currently it is not:
<img width="566" alt="image" src="https://user-images.githubusercontent.com/42802102/156619005-e90897f6-4ce7-4bd8-8f1a-ca9719aa1ef4.png">

I think this should be as simple as:
```suggestion
            disabled={isSchemaTitleInvalid}
```

An existing example where we're doing this would be in `CompleteSignupScreen.tsx`, I realise we aren't doing this everywhere atm.

Reply:
I assumed the required check on the input field will catch that edge case, but I get your point. Updated here => 6309ee3

Reply:
To be clear this suggestion wasn't about functionality, it was about having the button be visibly disabled (the styling should be different when it is disabled). I realise the user can't submit the form when the input is invalid thanks to the required field 👍